common-app-lib/src/cmr/common_app/services/kms_fetcher.clj

```diff
@@ -60,36 +62,50 @@
   "Calls GCMD KMS endpoints to retrieve the keywords. Response is a map structured in the same way
   as used in the KMS cache."
   [context]
-  (let [kms-cache (cache/context->cache context kms-cache-key)
-        _ (rl-util/log-redis-reading-start "kms-fetcher.fetch-gcmd-keywords-map" kms-cache-key)
-        [t1 kms-cache-value] (util/time-execution (cache/get-value kms-cache kms-cache-key))]
-    (rl-util/log-redis-read-complete "fetch-gcmd-keywords-map" kms-cache-key t1)
-    (kms-lookup/create-kms-index
-     context
-     (into {}
-           (for [keyword-scheme (keys kms/keyword-scheme->field-names)]
-             ;; if the keyword-scheme-value is nil that means we could not get the KMS keywords
-             ;; in this case use the cached value value instead so that we dont wipe out the cache.
-             (if-let [keyword-scheme-value (kms/get-keywords-for-keyword-scheme context keyword-scheme)]
-               [keyword-scheme keyword-scheme-value]
-               [keyword-scheme (get kms-cache-value keyword-scheme)]))))))
+  (try
+    (let [kms-cache (cache/context->cache context kms-cache-key)
+          _ (rl-util/log-redis-reading-start "kms-fetcher.fetch-gcmd-keywords-map" kms-cache-key)
+          [t1 kms-cache-value] (util/time-execution (cache/get-value kms-cache kms-cache-key))]
+      (rl-util/log-redis-read-complete "fetch-gcmd-keywords-map" kms-cache-key t1)
+      (kms-lookup/create-kms-index
+        context
+        (into {}
+              (for [keyword-scheme (keys kms/keyword-scheme->field-names)]
+                ;; if the keyword-scheme-value is nil that means we could not get the KMS keywords
+                ;; in this case use the cached value instead so that we don't wipe out the cache.
+                (if-let [keyword-scheme-value (kms/get-keywords-for-keyword-scheme context keyword-scheme)]
+                  [keyword-scheme keyword-scheme-value]
+                  [keyword-scheme (get kms-cache-value keyword-scheme)])))))
+    (catch Exception e
+      (if (clojure.string/includes? (ex-message e) "Carmine connection error")
+        (do
+          (error "fetch-gcmd-keywords-map found redis carmine exception. Will return nil result." e)
+          nil)
```

Comment:
Remove the (do    and the nil)
(error returns nil.

Please do this in the kms_lookup namespace as well.